src/rpc/rpc.cc

```diff
@@ -1176,12 +1176,13 @@ future<> server::connection::send_unknown_verb_reply(std::optional<rpc_clock_typ
       if (_options.streaming_domain) {
           _servers.erase(*_options.streaming_domain);
       }
+      _stop_watchdog.arm(std::chrono::minutes(1));
       return when_all(_ss_stopped.get_future(),
           parallel_for_each(_conns | boost::adaptors::map_values, [] (shared_ptr<connection> conn) {
               return conn->stop();
```

Comment:
I wonder if we need to capture `conn` around `conn->stop()`

Reply:
I think we don't. The server::connection::stop() doesn't sleep but only returns a future (by value) from connection::_stopped promise

Reply:
Right, but the `_stopped` promise goes poof if `connection` is destroyed...
It lifetime needs to be extended until `_stopped.set_value()` is called
and I'm afraid that https://github.com/scylladb/seastar/blob/0b9afec2e8f0f29b35801b88b202508704faf507/src/rpc/rpc.cc#L1081 might destroy the connection before that, so capturing it either on the set_value side or on the stop side will make that song and dance safer.

Reply:
The server::connection::process() keeps shared-from-this on its final `finally` continuation

https://github.com/scylladb/seastar/blob/0b9afec2e8f0f29b35801b88b202508704faf507/src/rpc/rpc.cc#L1090

so it still looks safe